src/macros-by-example.md

```diff
@@ -196,6 +196,220 @@ compiler knows how to expand them properly:
     not have the same number. This requirement applies to every layer of nested
     repetitions.
 
+## Metavariable expressions
+
+Gives access to additional information about metavariables that otherwise would be difficult or even impossible to obtain manually.
+
+### count($ident, depth)
+
+Expands to an unsuffixed integer literal representing the number of times a ***metavariable*** repeats in total.
+
+The output of `count` depends on where it is placed as well the provided index. If no index is provided, then it will always start at the innermost level.
+
+```rust
+macro_rules! count_idents {
+    ( $( $i:ident ),* ) => {
+        ${count($i)}
+    };
+}
+
+fn main() {
+    assert_eq!(count_idents!(a, b, c), 3);
+}
+```
+
+If repetitions are nested, then the optional depth parameter can be used to limit the number of nested repetitions that are counted.
+
+```rust
+macro_rules! no_repetition0 {
+    ( $( $a:ident: $( $b:literal ),* );+ ) => {
+        [${count($b)}]
+    };
+}
+
+macro_rules! no_repetition1 {
+    ( $( $a:ident: $( $b:literal ),* );+ ) => {
+        [${count($b, 1)}]
+    };
+}
+
+macro_rules! outermost {
+    ( $( $a:ident: $( $b:literal ),* );+ ) => {
+        [$( ${ignore($a)} ${count($b)}, )+]
+    };
+}
+
+fn main() {
+    // 1 2 3 4 5 = 5 elements
+    assert_eq!(no_repetition0!(a: 1, 2, 3; b: 4, 5), [5]);
+    
+    // a b = 2 elements
+    assert_eq!(no_repetition1!(a: 1, 2, 3; b: 4, 5), [2]);
+    
+    // 1 2 3 = 3 elements
+    // 4 5 = 2 elements
+    assert_eq!(outermost!(a: 1, 2, 3; b: 4, 5), [3, 2]);
+}
+```
+
+`count` can not be placed inside the innermost repetition.
```

Comment:
Would be good to have an example of that; I don't know what is meant here.

Is this even about the innermost repetition? In `( $( $name:ident: $( $value:literal ),* );+ )`, can I do `$(  ${count($name) }`? That is not the innermost repetition, as `value` is even-more-inner -- but I am trying to count `name` inside a repetition for `name` which will always return 1.

Reply:
> Is this even about the innermost repetition?

That is correct, `count` can not be placed inside the repetition depth of its referenced meta-variable. So `innermost` in regards to the macro declaration is inaccurate.

> can I do $(  ${count($name) }?

No because of the above explanation.

> but I am trying to count name inside a repetition for name which will always return 1.

That is the reason behind such constraint. It doesn't make sense to count something that will always yield `1`.

Nevertheless, it can be relaxed in the future if so desired.